rust-escrow/conditional-escrow/src/lib.rs

```diff
@@ -549,7 +549,116 @@ mod tests {
             "Withdrawal should be allowed"
         );
 
-        contract.delegate_funds();
+        contract.delegate_funds("dao1".to_string());
+    }
+
+    #[test]
+    #[should_panic(expected = "ERR_DELEGATE_NOT_ALLOWED")]
+    fn test_should_not_delegate_funds_if_already_delegated() {
+        let mut context = setup_context();
+
+        let expires_at = add_expires_at_nanos(100);
+
+        let mut contract = setup_contract(expires_at, MIN_FUNDING_AMOUNT);
+
+        testing_env!(context
+            .signer_account_id(bob())
+            .attached_deposit(MIN_FUNDING_AMOUNT / 2)
+            .build());
+
+        contract.deposit();
+
+        testing_env!(context
+            .signer_account_id(carol())
+            .attached_deposit(MIN_FUNDING_AMOUNT / 2)
+            .build());
+
+        contract.deposit();
+
+        testing_env!(context
+            .block_timestamp((expires_at + 200).try_into().unwrap())
+            .build());
+
+        assert_eq!(
+            false,
+            contract.is_deposit_allowed(),
+            "Deposit should not be allowed"
+        );
+
+        assert_eq!(
+            false,
+            contract.is_withdrawal_allowed(),
+            "Withdrawal should not be allowed"
+        );
+
+        contract.delegate_funds("dao1".to_string());
+
+        testing_env!(
+            context.build(),
+            near_sdk::VMConfig::test(),
+            near_sdk::RuntimeFeesConfig::test(),
+            Default::default(),
+            vec![PromiseResult::Successful("true".to_string().into_bytes())],
```

Comment:
Nice!

I was thinking, even that SputnikDAO is returning false. What if we in this contract, actually panicks. To be able to use the other `is_promise_successful` pattern?